app/src/main/java/org/fossasia/openevent/activities/SpeakersActivity.java

```diff
@@ -0,0 +1,98 @@
+package org.fossasia.openevent.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.fossasia.openevent.Adapters.SessionsAdapter;
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.dbutils.DbSingleton;
+import org.fossasia.openevent.utils.SpeakerIntent;
+
+import java.text.ParseException;
+import java.util.List;
+
+/**
+ * Created by MananWason on 30-06-2015.
+ */
+public class SpeakersActivity extends AppCompatActivity {
+    SessionsAdapter sessionsAdapter;
+    private String speaker;
+    private Speaker clicked;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_speakers);
+        DbSingleton dbSingleton = DbSingleton.getInstance();
+        speaker = getIntent().getStringExtra("SPEAKER");
+        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_speakers);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        try {
+            clicked = dbSingleton.getSpeakerbySpeakersname(speaker);
```

Comment:
Actually I am not getting the name from the db here. That would be very costly. I was just getting the text of the textview. So there's no way of getting the id without getting the db instane. Anyways, tracks have to be uniquely named, so, doesn't matter if I use the track name. What do you think?